File: frontend/app/graphql/fragments/Burger.ts
import {gql} from "@apollo/client";
import {AfspraakFragment} from "./Afspraak";
import {HuishoudenFragment} from "./Huishouden";
import {RekeningFragment} from "./Rekening";

export const BurgerFragment = gql`
    fragment Burger on Burger {
        id
        bsn
        email
        telefoonnummer
        voorletters
        voornamen
        achternaam
        geboortedatum
        straatnaam
        huisnummer
        postcode
        plaatsnaam
        rekeningen {
            ...Rekening
        }
        afspraken {
            ...Afspraak
        }
        huishouden {
            id
            burgers {
                id
            }
        }
    }
    ${HuishoudenFragment}
    ${AfspraakFragment}
    ${RekeningFragment}
`;

File: backend/hhb_backend/service/model/burger.py
from hhb_backend.service.model.base_model import BaseModel


class Burger(BaseModel):
    id: int
    uuid: str
    bsn: int
    saldo_alarm: bool
    start_date: int
    voorletters: str
    voornamen: str
    achternaam: str
    geboortedatum: str
    telefoonnummer: str
    email: str
    straatnaam: str
    huisnummer: str
    postcode: str
    plaatsnaam: str
    huishouden_id: int
    saldo: int
